platform-core-module/src/main/java/com/softicar/platform/core/module/test/container/TestContainerController.java

```diff
@@ -0,0 +1,208 @@
+import com.softicar.platform.common.io.command.ShellCommandExecutor;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.UUID;
+
+/**
+ * Manages startup and shutdown of a docker-containerized service for unit
+ * tests.
+ *
+ * @author Alexander Schmidt
+ */
+public class TestContainerController {
+
+	private final String containerName;
+	private final DockerCommandExecutor commandExecutor;
+	private boolean shutdownHookRegistered;
+	private String containerIpAddress;
+
+	/**
+	 * Constructs a new {@link TestContainerController}.
+	 *
+	 * @param containerNamePrefix
+	 *            a prefix for the names of the managed Docker containers; e.g.
+	 *            {@code "myservice-test"} (never <i>null</i>)
+	 * @param dockerImage
+	 *            the name of the Docker image or repository to start; e.g.
+	 *            {@code "myvendor/myservice:1.0.0"} or
+	 *            {@code "myvendor/myservice"} (assuming {@code ":latest"})
+	 *            (never <i>null</i>)
+	 */
+	public TestContainerController(String containerNamePrefix, String dockerImage) {
+
+		assertValidContainerNamePrefix(containerNamePrefix);
+		assertValidImageName(dockerImage);
+
+		this.containerName = "%s-%s".formatted(containerNamePrefix, UUID.randomUUID());
+		this.commandExecutor = new DockerCommandExecutor(containerName, dockerImage);
+		this.shutdownHookRegistered = false;
+		this.containerIpAddress = null;
+	}
+
+	public String getContainerIpAddress() {
+
+		return Optional//
+			.ofNullable(containerIpAddress)
+			.orElseThrow(() -> new RuntimeException("Failed to retrieve the IP address of the container: The container was not yet started."));
+	}
+
+	/**
+	 * Registers a shutdown hook in the current {@link Runtime}, to terminate
+	 * the container upon {@link Runtime} shutdown.
+	 *
+	 * @see #shutdown()
+	 */
+	public void registerRuntimeShutdownHook() {
+
+		if (!shutdownHookRegistered) {
+			Runtime.getRuntime().addShutdownHook(new Thread(this::shutdown));
+			this.shutdownHookRegistered = true;
+		}
+	}
+
+	/**
+	 * Terminates the container, and performs cleanup operations.
+	 * <p>
+	 * Invoked automatically, if a shutdown hook was registered.
+	 *
+	 * @see #registerRuntimeShutdownHook()
+	 */
+	public void shutdown() {
+
+		commandExecutor.executeContainerDown();
+	}
+
+	/**
+	 * Starts the container, and performs setup operations.
+	 */
+	public void startup() {
+
+		commandExecutor.executeContainerUp();
+		this.containerIpAddress = commandExecutor.executeContainerIpAddressRetrieval();
+	}
+
+	private void assertValidContainerNamePrefix(String containerNamePrefix) {
+
+		Objects.requireNonNull(containerNamePrefix);
+		if (!containerNamePrefix.matches("[a-zA-Z].*")) {
+			throw new IllegalArgumentException("Invalid container name prefix.");
+		}
+	}
+
+	private void assertValidImageName(String imageName) {
+
+		Objects.requireNonNull(imageName);
+		if (!imageName.matches("[a-zA-Z0-9].*/[a-zA-Z0-9].*")) {
```

Comment:
is that what you intended? it must only start with a letter or number? maybe add an inline comment, why this is

Reply:
yea that was even wrong. already updated.